qutebrowser/config/configtypes.py

```diff
@@ -1522,9 +1522,17 @@ class ShellCommand(List):
     _show_valtype = False
 
     def __init__(self, placeholder: bool = False,
+                 completions: _Completions = None,
                  none_ok: bool = False) -> None:
         super().__init__(valtype=String(), none_ok=none_ok)
         self.placeholder = placeholder
+        self._completions = completions
+
+    def complete(self) -> _Completions:
+        if self._completions is not None:
+            return self._completions
```

Comment:
coverage complains because this line isn't tested - as this will be refactored soon-ish anyways, feel free to add a `# pragma: no cover` for now.

Reply:
Also the line when the `default` file selector is used is not covered but not sure how to test that case
https://github.com/qutebrowser/qutebrowser/blob/965062c0b6800cfa3809f1171e7f8f7541626569/qutebrowser/browser/webengine/webview.py#L251

Reply:
Yep - there's no good way to test that automatically (I think I tried a while back, and ran into issues with not being able to close the file selectors, since they're not Qt widgets but native Windows/GTK/... things).

However, qutebrowser enforces 100% coverage for files which are typically easy to test (see `PERFECT_FILES` in `scripts/dev/check_coverage.py`). `configtypes.py` is one of those, but `webengine/webview.py` isn't, so this is needed here but not there.